File: BI-TJV/semestral/app-model/src/main/java/wrzecond/order/TjvOrderDTO.java
package wrzecond.order;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;

import java.sql.Timestamp;

@EqualsAndHashCode
@AllArgsConstructor(access = AccessLevel.PROTECTED)
abstract class TjvOrderDTO {

    // === PROPERTIES ===
    protected final Integer id;
    @Getter private final Integer table;
    @Getter private final Timestamp datetime;
    @Getter private final Boolean completed;

}

File: BI-KOT/semestral/app-model/src/main/kotlin/wrzecond/TjvOrderDTO.kt
package wrzecond

import java.sql.Timestamp

data class TjvOrderReadDTO (
    override val id: Int,
    val table: Int,
    val datetime: Timestamp,
    val completed: Boolean
) : IReadDTO { override val displayName = "$id" }

data class TjvOrderCreateDTO (
    val table: Int,
    val datetime: Timestamp,
    val completed: Boolean = false
) : ICreateDTO

data class TjvOrderUpdateDTO (
    val table: Int? = null,
    val datetime: Timestamp? = null,
    val completed: Boolean? = null
) : IUpdateDTO